R-Multiverse governance document

shikokuchuo opened the pull request:
This document is based loosely on the [Substrait governance document](https://substrait.io/governance/), but with important additions and differences.

I'll just highlight a few key points:

- Control over the project is maintained by the R-Multiverse Management Committee
- Decision making is by consensus, but a voting system is defined where this is not possible
- All communications to remain public
- Changes to policy by public proposal, which require one month notice period
- Definition of a 'pre-launch' period where that one month period is temporarily shortened to 2 weeks

The 4 different roles of User / Contributor / Moderator / Administrator are defined, and other policies such as the Review Policy and Code of Conduct can address each of these separately as required.

The spirit of the document is to include only what is minimally required, rather than being overly-prescriptive.

Having said that, let me know if you think other points should be covered here.

This document is here as it belongs on the website. I thought about adding it to the help repo or creating a copy there, but I'd rather have one source of truth. Any future modification proposals created at the help repo can link to PRs at this repo.

@wlandau @jeroen @maelle do feel free to comment. I will only merge this PR once all 4 of us are all in agreement.

maelle commented:
I plan to take a look at this next week. Thank you!!

maelle commented on governance.qmd:
```diff
@@ -0,0 +1,96 @@
+title: "R-Multiverse Governance"
+---
+
+The R-Multiverse project is run by volunteers in a collaborative and open way. Portions of this governance policy have been inspired by that of the Apache Software Foundation.
```
I'd really encourage adding a new line for each sentence, for nicer Git diff. :grin:

maelle commented on governance.qmd:
```diff
@@ -0,0 +1,96 @@
+title: "R-Multiverse Governance"
+---
+
+The R-Multiverse project is run by volunteers in a collaborative and open way. Portions of this governance policy have been inspired by that of the Apache Software Foundation.
```
What does volunteers mean? I'm asking since it can be a confusing term. People might be volunteers from a project's perspective, but still get paid by something else.

maelle replied:
so maybe add another word? or what is the role of this information?

shikokuchuo replied:
Agree, I don't think 'volunteer' adds any particular significance here, so I'll re-word this.

maelle commented on governance.qmd:
```diff
@@ -0,0 +1,96 @@
+
+### User
+
+A user is someone who uses R-Multiverse by downloading and installing packages from the R-Multiverse repositories. They may also contribute to R-Multiverse by providing feedback in the form of bug reports and feature suggestions in the `r-multiverse/help` repository. Users participate in the R-multiverse community by helping other users in `r-multiverse/help` or in other external forums.
```
I feel the user/contributor words are not very adapted since both categories can contribute. How about users vs developers? or package users vs package authors?

shikokuchuo replied:
I'll work on this section to delineate a bit more. The idea is that users are just anyone who may use R-Multiverse in `install.packages()`, whereas a contributor is someone who actively interacts with the R-Multiverse infrastructure by submitting a package. The obligations / expectations on the former are then much lower.

maelle commented on governance.qmd:
```diff
@@ -0,0 +1,96 @@
+---
+title: "R-Multiverse Governance"
```
should this live under github.com/r-multiverse in `.github/GOVERNANCE.md`, and be read from the qmd?

shikokuchuo replied:
I don't have a specific view on this, except that we shouldn't have a situation where the two can get out of sync.

maelle commented on governance.qmd:
```diff
@@ -0,0 +1,96 @@
+
+As illustration, if the total number of administrators is six, then at least four votes are required to pass a Proposal.
+
+## R-Multiverse Mission
```
should this come earlier?

shikokuchuo replied:
moved up.

maelle commented on governance.qmd:
```diff
@@ -0,0 +1,96 @@
+
+We invite third parties, including for-profit corporations, to contribute to R-Multiverse and use it for their own purposes. However it is important for the longevity and community health of R-Multiverse that the community be recognised as the primary contributor to the project and that the project continues to serve all members of the community equally.
+
+The RMC actively track the corporate allegiances of administrators and moderators, and strives to ensure influence around any particular aspect of the project is not overly skewed towards a single corporate entity.
```
allegiances? affiliations?

shikokuchuo replied:
yes, affiliations is a better choice of word :)

maelle commented on governance.qmd:
```diff
@@ -0,0 +1,96 @@
+- Charlie Gao (Hibiki AI)
+- Will Landau (Eli Lilly & Co.)
+- Jeroen Ooms (rOpenSci)
+- Maëlle Salmon (rOpensci)
```
should this track more of my allegiances :grin:

shikokuchuo replied:
Yes, will add Cynkra. Any others I should be aware of? I guess rOpenSci was top of mind here, as we already have two members and if there were another then the entire project could be 'captured'. This is part of what the 'tracking affiliations' section aims to mitigate against.

maelle reviewed (commented):
thank you, this looks good!

shikokuchuo commented:
Thanks @maelle I've addressed your review comments in [dd2c36c](https://github.com/r-multiverse/r-multiverse.github.io/pull/16/commits/dd2c36cdbad7af9409233c4c5eb9563a3bd58c05).

The only question outstanding is whether this document should live in a specific location. I am happy to have the website (public) version be the source of truth. As I mentioned in a previous issue, I don't think GitHub itself recognizes the governance document even if we put it in a specific location, so I'm not sure it adds anything.

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,117 @@
+# The R-Multiverse Project
+
+The R-Multiverse project consists of the code and repositories that reside in the `r-multiverse` GitHub organization and the r-multiverse.org website.
+All repositories are open to everyone, with the exception of a private repository where security and conduct-related matters are shared between the administrators only.
```
Maybe "All repositories are visible to everyone" might be clearer than "All repositories are open to everyone".

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,117 @@
+
+### Communication
+
+Communication is always done publicly via GitHub.
```
except for confidential security and conduct-related matters.

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,117 @@
+### Communication
+
+Communication is always done publicly via GitHub.
+There are no private mailing lists and all decisions related to the project are made in public.
```
Maybe mention that the private repo makes confidential conversations more traceable and able to be audited than a mailing list.

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,117 @@
+The R-Multiverse Management Committee (RMC) is responsible for the active management of the R-Multiverse project.
+The main role of the RMC is to further the long-term development and health of the community as a whole, and to ensure that balanced and broad-based user engagement and contributor participation takes place.
+
+As part of this, the RMC is the primary proposer and approver of changes to this governance document and all policies regarding operation of the `r-multiverse` GitHub repositories.
```
Can community members put forth proposals too? Or just admins?

shikokuchuo replied:
Have re-worded to make this unambiguous: "the RMC is the ~~primary~~ __sole__ proposer and approver of changes"

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,117 @@
+All members of the community are invited to participate, and Proposals may be withdrawn or revised by their initial proposer as a result.
+Revisions likely to be regarded as material shall be proposed as a new Proposal, with reference to the preceding discussion.
+
+Proposals will be deemed to have been adopted if both one calendar month has passed after notice has been publicly posted and either (i) there has been no further response from an administrator or (ii) dissent by an administrator was recorded and the Proposal has been passed after voting has taken place as per the Voting Process described below.
```
What if a community member posts a proposal and no admins get around to reading it? After 1 month, if no admin says no, is the proposal automatically passed?

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,117 @@
+
+### R-Multiverse Management Committee (RMC)
+
+Name in alphabetical order (affiliation)
```
alphabetical by last name

shikokuchuo replied:
I think it's fine to leave as is (from a business/legal context). The significance of the text is to say there is no seniority implied by the ordering. Logically it then follows that it doesn't matter what that actually is.

wlandau reviewed (changes requested):
Excellent job overall! I just have some minor comments.

As for the location, I would prefer the website because I think it is more discoverable. I would prefer it to be the top entry of the "Policies" dropdown: https://github.com/r-multiverse/r-multiverse.github.io/blob/437b050ce13caeb89d0b01cb2c4f149b5953f93b/_quarto.yml#L22-L25

shikokuchuo commented:
Thanks for the review @wlandau. I've addressed your comments in [c9802bb](https://github.com/r-multiverse/r-multiverse.github.io/pull/16/commits/c9802bb1236f4dd324f04cb9ca7737ca25eb893a) and also added the governance document to the website navigation as suggested.

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,119 @@
+All members of the community are invited to participate, and Proposals may be withdrawn or revised by their initial proposer as a result.
+Revisions likely to be regarded as material shall be proposed as a new Proposal, with reference to the preceding discussion.
+
+Proposals will be deemed to have been adopted if both one calendar month has passed after notice has been publicly posted and either (i) there has been no further response from an administrator or (ii) dissent by an administrator was recorded and the Proposal has been passed after voting has taken place as per the Voting Process described below.
```
Maybe it would be clearer to say "(i) there has been no dissent from an administrator" instead of "(i) there has been no further response from an administrator". Otherwise, the trivial case of explicit unanimity seems to fall through the cracks as worded.

shikokuchuo replied:
Good catch, thanks! Amended in [7467dce](https://github.com/r-multiverse/r-multiverse.github.io/pull/16/commits/7467dce293fc1b359f5e3d1bb017d36d8da4ba8e).

wlandau commented:
Thanks, @shikokuchuo. I approve, with one final comment at https://github.com/r-multiverse/r-multiverse.github.io/pull/16/files#r1633763887.

How do you propose we implement proposals? It seems reasonable to add a new thread at https://github.com/r-multiverse/help/issues with label "proposal", but then moderators might have the power to change the label if we give them "triage" permissions on GitHub.

wlandau commented:
Which actually brings up a potential new role: "community moderator", or just "moderator". We could give these people "triage" access to all the discussion forums, etc. This role could be different from the existing "package monitor" / "reviewer" role of on-boarding packages at https://github.com/r-multiverse/contributions and resolving disputes. Alternatively, everything could go under the monolithic role of "moderator", or admins could take sole responsibility for community outreach. What do you think?

shikokuchuo commented:
> How do you propose we implement proposals? It seems reasonable to add a new thread at https://github.com/r-multiverse/help/issues with label "proposal", but then moderators might have the power to change the label if we give them "triage" permissions on GitHub.

Yes exactly, just using the tag and probably pinning them would be good. I think we can trust the moderators not to touch the labels, and not require anything more than that.

shikokuchuo commented:
> Which actually brings up a potential new role: "community moderator", or just "moderator". We could give these people "triage" access to all the discussion forums, etc. This role could be different from the existing "package monitor" / "reviewer" role of on-boarding packages at https://github.com/r-multiverse/contributions and resolving disputes. Alternatively, everything could go under the monolithic role of "moderator", or admins could take sole responsibility for community outreach. What do you think?

I don't think we need to be so granular as to have different types of moderator. Both admins and moderators should be able to do community outreach. I think that would be inherent in the moderator role in any case for questioning packages etc., but they could also proactively look to curate their domain of expertise.

wlandau commented:
> I don't think we need to be so granular as to have different types of moderator. Both admins and moderators should be able to do community outreach. 

In that case, I prefer to be pedantic and direct about this in the governance document. Could the "moderator" section grant them explicit authority to moderate community forums? It would also help to explicitly clarify that admins have all the privileges of moderators.

shikokuchuo commented:
I've added this in [6d4e3cd](https://github.com/r-multiverse/r-multiverse.github.io/pull/16/commits/6d4e3cd86acc00c43eab1e6709ab99e40cc1139b).

We don't need to explicitly mention the second point (and to void the 'inheritance', a point you mentioned previously) as it is alredy clearly stated that "The RMC collectively appoint and delegate authority to moderators to perform a limited subset of duties."

wlandau commented:
> In that case, I prefer to be pedantic and direct about this in the governance document. Could the "moderator" section grant them explicit authority to moderate community issue trackers and discussion forums?

Thanks for adding this, I have no more suggestions.

> It would also help to explicitly clarify that admins have all the privileges of moderators.

On second thought, an admin who moderates is really both an admin and a moderator. I like keeping these roles separate for the sake of having clean and clear definitions.

maelle commented on governance.qmd:
```diff
@@ -0,0 +1,120 @@
+- Charlie Gao (Hibiki AI)
+- Will Landau (Eli Lilly & Co.)
+- Jeroen Ooms (rOpenSci)
+- Maëlle Salmon (cynkra, rOpensci)
```
```suggestion
- Maëlle Salmon (rOpensci, cynkra)
```

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,120 @@
+title: "R-Multiverse Governance"
+---
+
+The R-Multiverse project is run in a collaborative and open way.
```
As Noam mentioned today, it would be good to expand on this section to include:

1. The mission, values, and purpose of R-multiverse.
2. The scope and purpose of this governance document.

wlandau commented on governance.qmd:
```diff
@@ -0,0 +1,120 @@
+
+Name in alphabetical order (affiliation)
+
+- Charlie Gao (Hibiki AI)
```
The names of admins are hard-coded into the document, so implicitly changes to the document constitute changes to the RMC. However, this may not be obvious at first glance. Would be nice to clarify this in the "Proposals" section.

shikokuchuo commented:
In commit [5a1ec37](https://github.com/r-multiverse/r-multiverse.github.io/pull/16/commits/5a1ec372bcd2b3f5c5ce4e434607fe568b4b684f) above, I update the document post our discussion today.

Specifically:

- Purpose of document add to first paragraph
- R-Multiverse Mission section moved up front
- Includes subsection that explicitly sets out that the document is expected to evolve with needs over time
- I have removed the "pre-launch" period that was in there previously as it seems this document should be in fairly good shape to begin with, and it would be good practice to start off as we mean to continue
- Added subsection explicitly mentioning the policy for changes to the RMC
- Added 'language' section at end

shikokuchuo commented:
I've re-requested review from all. @noamross thanks also for your inputs today. Very helpful in moving this along.

wlandau reviewed (changes requested):
Looks great, just one minor grammatical note inline.

shikokuchuo commented on governance.qmd:
```diff
@@ -0,0 +1,132 @@
+# The R-Multiverse Project
+
+The R-Multiverse project consists of the code and repositories that reside in the `r-multiverse` GitHub organization and the r-multiverse.org website.
+All repositories are public, and transparent and visible for everyone, with the exception of a private repository where security and conduct-related matters are shared between the administrators only.
```
Adopted in [ddb9779](https://github.com/r-multiverse/r-multiverse.github.io/pull/16/commits/ddb977912fb243da3098b2abb8bcc85430ac80c4). This is very subtle in terms of nuance, and I tend to agree, but only just...

shikokuchuo commented:
I've just moved the contents of the document into a plain .md file for easier viewing on GitHub (as suggested by @eitsupi).

@maelle and @jeroen for reviewing the changes post our most recent call please refer to commit [5a1ec37](https://github.com/r-multiverse/r-multiverse.github.io/pull/16/commits/5a1ec372bcd2b3f5c5ce4e434607fe568b4b684f). Please let me know if you have any further comments on these. Thanks.

maelle commented on _quarto.yml:
```diff
@@ -3,6 +3,7 @@ project:
   output-dir: docs
   render:
     - "*.qmd"
+    - "governance.md"
```
one thing, why not just make it qmd?

shikokuchuo replied:
was a suggestion by @eitsupi - so people can view via GitHub with good formatting

jeroen commented:
Looks good to me, thanks!
